### ISSUE TITLE
Set specific version in cpanfile fail (DBD::Pg example)

### ISSUE THREAD

**Issue by avkhozov** [miyagawa/cpanminus#440]
I can install ``DBD::Pg`` any version via cpanm, but I can't when version is specified in ``cpanfile``.

```text
# Check 3.0.0 (OK)
$ sudo cpanm -n --mirror http://cpan.org/ --mirror-only TURNSTEP/DBD-Pg-3.0.0.tar.gz
--> Working on TURNSTEP/DBD-Pg-3.0.0.tar.gz
Fetching http://cpan.org/authors/id/T/TU/TURNSTEP/DBD-Pg-3.0.0.tar.gz ... OK
Configuring DBD-Pg-3.0.0 ... OK
Building DBD-Pg-3.0.0 ... OK
Successfully installed DBD-Pg-3.0.0
1 distribution installed

# Check 3.2.1 (OK)
$ sudo cpanm -n --mirror http://cpan.org/ --mirror-only TURNSTEP/DBD-Pg-3.2.1.tar.gz
--> Working on TURNSTEP/DBD-Pg-3.2.1.tar.gz
Fetching http://cpan.org/authors/id/T/TU/TURNSTEP/DBD-Pg-3.2.1.tar.gz ... OK
Configuring DBD-Pg-3.2.1 ... OK
Building DBD-Pg-3.2.1 ... OK
Successfully installed DBD-Pg-3.2.1
1 distribution installed
 
# Set 3.2.1 in cpanfile
$ cat cpanfile
requires 'DBD::Pg', '== 3.2.1';
 
# OK
$ sudo cpanm -n --mirror http://cpan.org/ --mirror-only --installdeps .
--> Working on .
Configuring /home/and/projects/mprove ... OK
<== Installed dependencies for .. Finishing.
 
# Set 3.0.0 in cpanfile
$ cat cpanfile
requires 'DBD::Pg', '== 3.0.0';
 
# Fail
$ sudo cpanm -n --mirror http://cpan.org/ --mirror-only --installdeps .
--> Working on .
Configuring /home/and/projects/mprove ... OK
==> Found dependencies: DBD::Pg
Found DBD::Pg 3.5.1 which doesn't satisfy == v3.0.0.
! Installing the dependencies failed: Installed version (3.2.1) of DBD::Pg is not in range '== v3.0.0'
! Bailing out the installation for ..
 
# Set ragne for 3.0.0
$ cat cpanfile
requires 'DBD::Pg', '>= 3.0.0, < 3.2.1';
 
# Fail
$ sudo cpanm -n --mirror http://cpan.org/ --mirror-only --installdeps .
--> Working on .
Configuring /home/and/projects/mprove ... OK
==> Found dependencies: DBD::Pg
Found DBD::Pg 3.5.1 which doesn't satisfy >= v3.0.0, < v3.2.1.
! Installing the dependencies failed: Installed version (3.2.1) of DBD::Pg is not in range '>= v3.0.0, < v3.2.1'
! Bailing out the installation for ..
```

**Comment by avkhozov** [comment]
cpanm version:
```text
$ sudo cpanm -V
cpanm (App::cpanminus) version 1.7027 (/usr/local/bin/cpanm)
perl version 5.018002 (/usr/bin/perl)

  %Config:
    archname=x86_64-linux-gnu-thread-multi
    installsitelib=/usr/local/share/perl/5.18.2
    installsitebin=/usr/local/bin
    installman1dir=/usr/share/man/man1
    installman3dir=/usr/share/man/man3
    sitearchexp=/usr/local/lib/perl/5.18.2
    sitelibexp=/usr/local/share/perl/5.18.2
    vendorarch=/usr/lib/perl5
    vendorlibexp=/usr/share/perl5
    archlibexp=/usr/lib/perl/5.18
    privlibexp=/usr/share/perl/5.18
  %ENV:
  @INC:
    FatPacked::15484944=HASH(0xec4810)
    /etc/perl
    /usr/local/lib/perl/5.18.2
    /usr/local/share/perl/5.18.2
    /usr/lib/perl5
    /usr/share/perl5
    /usr/lib/perl/5.18
    /usr/share/perl/5.18
    /usr/local/lib/site_perl
    .
```

**Comment by miyagawa** [comment]
Cannot pinpoint specifically but i'm sure this is a problem on metacpan regarding x.y.z version string format.

**Comment by miyagawa** [comment]
this should be fixed with 1.7033.
